src/esphomelib/uart_component.cpp

```diff
@@ -115,11 +120,22 @@ void UARTComponent::flush() {
 #ifdef ARDUINO_ARCH_ESP8266
 void UARTComponent::setup() {
   ESP_LOGCONFIG(TAG, "Setting up UART bus...");
+  // Use Arduino HardwareSerial UARTs if all used pins match the ones
+  // preconfigured by the platform. For example if RX disabled but TX pin
+  // is 1 we still want to use Serial.
   if (this->tx_pin_.value_or(1) == 1 && this->rx_pin_.value_or(3) == 3) {
-    // use UART0 if all used pins match the ones on the UART bus
-    // for example if RX disabled but TX pin is 1 we still want to use Serial.
     this->hw_serial_ = &Serial;
     this->hw_serial_->begin(this->baud_rate_);
+    this->hw_serial_->setDebugOutput(false);
```

Comment:
I do not think we should disable debug output here. If the user doesn't want debug output on a serial port they should disable the logger component for it. 

If they don't, even though we disable the SDKs debug output here, the user will still get debug output through esphome's own logging engine.

Reply:
I'll re-test without this but with the platformio build flag. My experience was that I still got debug output on UART0 unless I called setDebugOutput(false) on it here. This is due to the Arduino framework enabling it by default even before the esphome logger takes over. The esphome logger calls setDebugOutput(true) on the uart selected for logging, so the change is just to put debugging on the logger UART, not to disable it entirely.

Reply:
I guess that in the case of UART and Logger sharing a port, the UART would disable it after the Logger enabled it, so you're right - this isn't safe to do unconditionally.